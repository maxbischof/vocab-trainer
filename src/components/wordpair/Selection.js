import React from 'react'
import styled from 'styled-components'

export default function Selection({ text, active, color, diameter, onClick }) {
  const activeColor = color.substring(0, color.length - 2) + '0.44)'
  const borderSize = (parseInt(diameter) * 1.5 - parseInt(diameter)) / 2
  const fontSize = parseInt(diameter) / 2

  return (
    <Circle
      active={active}
      color={color}
      activeColor={activeColor}
      diameter={diameter}
      borderSize={borderSize}
      fontSize={fontSize}
      onClick={() => onClick(text)}
    >
      {text}
    </Circle>
  )
}

const Circle = styled.div`
  height: ${(props) => props.diameter};
  width: ${(props) => props.diameter};
  margin: 5px 0;
  background-color: ${(props) => props.color};
  border-radius: 50%;
  display: inline-block;
  display: grid;
  align-content: center;
  justify-content: center;
  font-weight: 700;
  font-size: ${(props) => props.fontSize + 'px'};
  background-clip: padding-box;
  border: ${(props) =>
    props.active
      ? props.borderSize + 'px solid ' + props.activeColor + ';'
      : ''};
  cursor: pointer;
`
