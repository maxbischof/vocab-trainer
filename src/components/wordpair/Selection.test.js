import React from 'react'
import Selection from './Selection'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<Selection text="A" active={true} color="rgba(43, 173, 134, 1)" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
