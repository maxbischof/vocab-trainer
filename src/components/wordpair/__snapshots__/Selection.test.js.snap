// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ buhiFj"
  color="rgba(43, 173, 134, 1)"
>
  A
</div>
`;
